rc/tools/git.kak

```diff
@@ -326,3 +337,124 @@ define-command -params 1.. \
+        # for -,+,<space>
+        set-option global git_diff_cursor_column %sh{ echo $(($kak_cursor_column-1)) }
+
+        # Allowed to press <ret> only _within_ a hunk
+        # Note: - Pressing <ret> on a @@ line (i.e. a "range line") is allowed.
+        #         This action will result in navigating to section heading (see below).
+        #       - Pressing <ret> on a +++ line is allowed. This action will result in
+        #         navigating to the first line of the file
+        execute-keys -draft 'x<a-K>^\w<ret>'
+        execute-keys -draft 'x<a-K>^$<ret>'
+        execute-keys -draft 'x<a-K>^---<ret>'
+
+        # Find the source filename for the current hunk (reverse search)
+        evaluate-commands -draft %{
+            execute-keys 'x<semicolon><a-/>^\+\+\+ b([^\n]*)<ret>'
+            set-option global git_diff_hunk_filename %reg{1}
+        }
+
+        try %{
+            # Are we on a +++ line? If so simply go to the first line of the file
+            execute-keys -draft 'x<a-k>^\+\+\+<ret>'
+            # Goto line 1 as we pressed on +++
+            edit -existing "%opt{git_diff_git_dir}%opt{git_diff_hunk_filename}" 1
+        } catch %{
+            # Find the source line at which the current hunk starts (reverse search)
+            evaluate-commands -draft %{
+                execute-keys 'x<semicolon><a-/>^@@ -\d+,\d+ \+(\d+),\d+ @@<ret>'
+                set-option buffer git_diff_hunk_line_num_start %reg{1}
+            }
+            # If we're already on a range line (i.e. a line that starts with @@) when
+            # <ret> was pressed our behavior slightly alters: we need to go look for the section heading.
+            # For example for range line:  @@ -123,4 +123,4 @@ fn some_function_name_possibly
+            # Here the section heading is "fn some_function_name_possibly". Please note that the section
+            # heading is NOT at line range line number necessarily so we can't trivially extract that.
+            try %{
+                # First things first, are we on a range line? If not, head to the nearest `catch`
+                execute-keys -draft 'x<a-k>^@@<ret>'
+                # Sane default. This is where we will go if:
+                # (a) section heading could not be found for some reason (b) section heading does not exist
+                set-option global git_diff_go_to_line_num %opt{git_diff_hunk_line_num_start}
+
+                try %{
+                    # Now search for the section heading!
+                    #
+                    # We're doing something a bit weird here: we're editing the file in a draft context!
+                    # The reason is that first we open up the file and _then_ we try to navigate to the
+                    # section heading. This adds two locations to the jump list. We ideally should have gone
+                    # to the section heading in one step but didn't know its line number in advance.
```

Comment:
This would indicate a bug in Kakoune, it should collapse multiple jumps done in `evaluate-commands` to a single one.